bindings/ocaml/Hacl.mli

```diff
@@ -405,50 +405,68 @@ module NaCl : sig
       (** [box_open ct n pk sk pt] attempts to verify and decrypt ciphertext [ct] using public key [pk],
           secret key [sk], and nonce [n] and if successful writes the plaintext in [pt]
           and returns true. *)
 
       (** {2 Precomputation interface }
           The shared key [ck] is obtained using {!NaCl.box_beforenm} or {!NaCl.Noalloc.box_beforenm}. *)
 
-      val box_afternm : pt:bytes -> n:bytes -> ck:bytes -> ct:bytes -> bool
+      val box_afternm : pt:bytes -> ?pt_offset:int -> ?pt_len:int -> n:bytes -> ck:bytes -> ct:bytes -> ?ct_offset:int -> unit -> bool
       (** [box_afternm pt n ck ct] authenticates and encrypts [pt] using shared key [ck] and
           nonce [n] and writes both the message authentication tag and the ciphertext in [ct].
           Returns true if successful. *)
 
-      val box_open_afternm : ct:bytes -> n:bytes -> ck:bytes -> pt:bytes -> bool
+      val box_open_afternm : ct:bytes -> ?ct_offset:int -> ?ct_len:int -> n:bytes -> ck:bytes -> pt:bytes -> ?pt_offset:int -> unit -> bool
       (** [box_open ct n pk sk pt] attempts to verify and decrypt ciphertext [ct] using
           shared key [ck] and nonce [n] and if successful writes the plaintext in [pt]
           and returns true. *)
 
       (** {1 Secretbox} *)
 
-      val secretbox : pt:bytes -> n:bytes -> key:bytes -> ct:bytes -> bool
+      val secretbox : pt:bytes -> ?pt_offset:int -> ?pt_len:int -> n:bytes -> key:bytes -> ct:bytes -> ?ct_offset:int -> unit -> bool
       (** [secretbox pt n key ct] authenticates and encrypts plaintext [pt] using
           secret key [key] and nonce [n] and writes both the message authentication tag
           and the ciphertext in [ct].
           Returns true if successful. *)
 
-      val secretbox_open : ct:bytes -> n:bytes -> key:bytes -> pt:bytes -> bool
+      val secretbox_open : ct:bytes -> ?ct_offset:int -> ?ct_len:int -> n:bytes -> key:bytes -> pt:bytes -> ?pt_offset:int -> unit -> bool
       (** [secretbox_open ct n key pt] attempts to verify and decrypt ciphertext [ct] using
           secret key [key] and nonce [n] and if successful writes the plaintext in [pt]
           and returns true. *)
     end
-    (** The {i easy} interface concatenates the ciphertext and the 16-byte long message
-        authentication tag into a single buffer.
+    (** The {i easy} interface concatenates the ciphertext and the 16-byte long
+        message authentication tag into a single buffer.
+
+        By default, these functions use the whole of [pt] and [ct], but users
+        can choose to only pass portions of these buffers, by passing some of
+        these optional arguments:
+        - [pt_offset], [ct_offset]: start at the specified position in [pt] or
+          [ct] (0 by default)
+        - [pt_len] or [ct_len]: specify the number of bytes to take in [pt] or
```

Comment:
is it implicit that if pt_offset is specified, then pt_len is taken to be "all the way to the end of the buffer"?

Reply:
It's always the case (whether `pt_offset` is specified or not) that not specifying `pt_len` means "all the way to the end".